trunk/wc_yotpo.php

```diff
@@ -127,6 +133,53 @@ function wc_yotpo_show_qna_widget() {
 		echo generate_v3_qna_widget_code($product, $qna_widget_id);
 	}						
 }
+// PROMOTED PRODUCTS WIDGET
+function wc_yotpo_show_promoted_products_widget() {
+	global $product;
+	$yotpo_settings = get_option('yotpo_settings',wc_yotpo_get_default_settings());
+	$promoted_products_widget_id = $yotpo_settings['v3_widgets_ids']['promoted_products'];
+	if (!$promoted_products_widget_id) {
+		return;
+	}
+	if(is_product() && $product->get_reviews_allowed() == true) {
+		$product_data = wc_yotpo_get_product_data($product);
+		echo generate_v3_promoted_products_widget_code($promoted_products_widget_id, $product_data['id']);
+	} else {
+		echo generate_v3_promoted_products_widget_code($promoted_products_widget_id);
+	}
+}
+// REVIEWS CAROUSEL WIDGET
+function wc_yotpo_show_reviews_carousel_widget(): void {
+	global $product;
+	$yotpo_settings = get_option('yotpo_settings',wc_yotpo_get_default_settings());
+	$reviews_carousel_widget_id = $yotpo_settings['v3_widgets_ids']['reviews_carousel'];
+	if (!$reviews_carousel_widget_id) {
+		return;
+	}
+	if(is_product() && $product->get_reviews_allowed() == true) {
+		$product_data = wc_yotpo_get_product_data($product);
+		echo generate_v3_reviews_carousel_widget_code($reviews_carousel_widget_id, $product_data['id']);
+	} else {
+		echo generate_v3_reviews_carousel_widget_code($reviews_carousel_widget_id);
+	}
+}
+// REVIEWS TAB WIDGET
+function wc_yotpo_show_reviews_tab_widget() {
+	global $product;
+	$yotpo_settings = get_option('yotpo_settings',wc_yotpo_get_default_settings());
+	$reviews_tab_widget_id = $yotpo_settings['v3_widgets_ids']['reviews_tab'];
+	if($product->get_reviews_allowed() == true && $reviews_tab_widget_id) {
+		echo generate_v3_reviews_tab_widget_code($product, $reviews_tab_widget_id);
+	}						
+}
+// CUSTOM WIDGETS
+function wc_yotpo_show_custom_widgets() {
+	global $product;
+	if($product->get_reviews_allowed() == true) {
+		foreach (generate_v3_custom_widgets_code($product) as $widget_code);
+		echo $widget_code;
```

Comment:
i'm not familiar with that syntax. is it correct?

Reply:
it was used like that before, and AFAIK that's the way PHP is being used